### PR TITLE
New function plot_with_gradient

### PR DESCRIPTION
**Pull request by mellamansanchez**
Used to plot a function with a gradient of colors depending of a list of values, e.g. depth

### CODE REVIEW

**Comment by vnmabus** [comment]
I do not know why the diff reports everything as changed. Maybe you changed newlines to \r\n? Please, use only \n as the line separator.

**Comment by mellamansanchez** [comment]
> I do not know why the diff reports everything as changed. Maybe you changed newlines to \r\n? Please, use only \n as the line separator.

I dont really know what could have happened I will review it, but I dont really understand what that change means.

**Comment by vnmabus** [comment]
> 
> 
> > I do not know why the diff reports everything as changed. Maybe you changed newlines to \r\n? Please, use only \n as the line separator.
> 
> I dont really know what could have happened I will review it, but I dont really understand what that change means.

The convention for representing new lines is different in different domains and Operating Systems. Unix systems tend to use the character LF (line feed or '\n') to represent the end of a line, while Windows and some Internet protocols used CRLF (carriage return plus line feed or '\r\n'). This is usually configurable in your editor.

We use always the Unix convention, so please configure your editor to follow that.

**Comment by mellamansanchez** [comment]
> > > I do not know why the diff reports everything as changed. Maybe you changed newlines to \r\n? Please, use only \n as the line separator.
> > 
> > 
> > I dont really know what could have happened I will review it, but I dont really understand what that change means.
> 
> The convention for representing new lines is different in different domains and Operating Systems. Unix systems tend to use the character LF (line feed or '\n') to represent the end of a line, while Windows and some Internet protocols used CRLF (carriage return plus line feed or '\r\n'). This is usually configurable in your editor.
> 
> We use always the Unix convention, so please configure your editor to follow that.

Ok thanks, I will review it.

File: skfda/exploratory/visualization/representation.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def _get_color_info(fdata, group, group_names, group_colors, legend, kwargs):
             sample_colors = fdata.n_samples * [kwargs.get("c")]
             kwargs.pop('c')
 
-        else:
```
Comment (vnmabus) [inline skfda/exploratory/visualization/representation.py:74]: Please configure your editor to remove trailing whitespace.

File: skfda/exploratory/visualization/representation.py [hunk 2]
```diff
@@ -85,9 +85,8 @@ def plot_graph(fdata, chart=None, *, fig=None, axes=None,
                **kwargs):
     """Plot the FDatGrid object graph as hypersurfaces.
 
-    Plots each coordinate separately. If the :term:`domain` is one dimensional,
-    the plots will be curves, and if it is two dimensional, they will be
-    surfaces.
+    Plots each coordinate separately. If the domain is one dimensional, the
```
Comment (vnmabus) [inline skfda/exploratory/visualization/representation.py:88]: This has been changed in the develop branch, and the changes should not be reverted.

File: skfda/exploratory/visualization/representation.py [hunk 3]
```diff
@@ -155,7 +154,6 @@ def plot_graph(fdata, chart=None, *, fig=None, axes=None,
         fdata, group, group_names, group_colors, legend, kwargs)
 
     if fdata.dim_domain == 1:
-
```
Comment (vnmabus) [inline skfda/exploratory/visualization/representation.py:158]: Why are you removing blank lines?

File: skfda/exploratory/visualization/representation.py [hunk 4]
```diff
@@ -323,3 +320,165 @@ def plot_scatter(fdata, chart=None, *, grid_points=None,
...
+                    color_dict["color"] = sample_colors[j]
+
+                axes[i].plot_surface(X, Y, Z[j, ..., i],
+                                     **color_dict, **kwargs)
+
+    _set_labels(fdata, fig, axes)
+
+    return fig
+
+def dd_plot(fdata, dist1, dist2, depth_method, chart=None, *, fig=None,
```
Comment (vnmabus) [inline skfda/exploratory/visualization/representation.py:459]: This functionality belong to a separate PR.

File: skfda/exploratory/visualization/representation.py [hunk 5]
```diff
@@ -323,3 +320,165 @@ def plot_scatter(fdata, chart=None, *, grid_points=None,
     _set_labels(fdata, fig, axes, patches)
 
     return fig
+
+def plot_with_gradient(fdata, chart=None, *, fig=None, axes=None,
```
Comment (vnmabus) [inline skfda/exploratory/visualization/representation.py:324]: We will try to follow the scikit-learn visualization API, with *Display objects and class methods to create them.

**Review by vnmabus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**